src/pages/resultsView/plots/PlotsTab.tsx

```diff
@@ -0,0 +1,1110 @@
+            },
+            set molecularProfileId(id:string|undefined) {
+                this._molecularProfileId = id;
+            },
+            get clinicalAttributeId() {
+                if (this._clinicalAttributeId === undefined && self.clinicalAttributeOptions.length) {
+                    return self.clinicalAttributeOptions[0].value;
+                } else {
+                    return this._clinicalAttributeId;
+                }
+            },
+            set clinicalAttributeId(id:string|undefined) {
+                this._clinicalAttributeId = id;
+            },
+            get logScale() {
+                return this._logScale && logScalePossible(this);
+            },
+            set logScale(v:boolean) {
+                this._logScale = v;
+            },
+            _axisType: AxisType.molecularProfile,
+            _entrezGeneId: undefined,
+            _molecularProfileType: undefined,
+            _molecularProfileId: undefined,
+            _clinicalAttributeId: undefined,
+            _logScale: true
+        });
+    }
+
+    private onInputClick(event:React.MouseEvent<HTMLInputElement>) {
+        switch (parseInt((event.target as HTMLInputElement).value, 10)) {
+            case EventKey.horz_molecularProfile:
+                this.horzSelection.axisType = AxisType.molecularProfile;
+                break;
+            case EventKey.horz_clinicalAttribute:
+                this.horzSelection.axisType = AxisType.clinicalAttribute;
+                break;
+            case EventKey.vert_molecularProfile:
+                this.vertSelection.axisType = AxisType.molecularProfile;
+                break;
+            case EventKey.vert_clinicalAttribute:
+                this.vertSelection.axisType = AxisType.clinicalAttribute;
+                break;
+            case EventKey.horz_logScale:
+                this.horzSelection.logScale = !this.horzSelection.logScale;
+                break;
+            case EventKey.vert_logScale:
+                this.vertSelection.logScale = !this.vertSelection.logScale;
+                break;
+            case EventKey.utilities_viewCopyNumber:
+                this.viewType = ViewType.CopyNumber;
+                break;
+            case EventKey.utilities_viewMutationType:
+                this.viewType = ViewType.MutationType;
+                break;
+        }
+    }
+
+    private setSearchCaseInput(e:any) {
+        this.searchCaseInput = e.target.value;
+        clearTimeout(this.searchCaseTimeout);
+        this.searchCaseTimeout = setTimeout(()=>this.executeSearchCase(this.searchCaseInput), searchInputTimeoutMs);
+    }
+
+    private setSearchMutationInput(e:any) {
+        this.searchMutationInput = e.target.value;
+        clearTimeout(this.searchMutationTimeout);
+        this.searchMutationTimeout = setTimeout(()=>this.executeSearchMutation(this.searchMutationInput), searchInputTimeoutMs);
+    }
+
+    private executeSearchCase(caseId:string) {
```

Comment:
i think these should all be actions since they are setting an observable

Reply:
done